src/main/groovy/com/google/protobuf/gradle/GenerateProtoTask.groovy

```diff
@@ -203,10 +206,37 @@ public abstract class GenerateProtoTask extends DefaultTask {
   }
 
   static int getCmdLengthLimit(String os) {
-    if (os != null && os.toLowerCase(Locale.ROOT).indexOf("win") > -1) {
-      return WINDOWS_CMD_LENGTH_LIMIT
+    return isWindows(os) ? WINDOWS_CMD_LENGTH_LIMIT : Integer.MAX_VALUE
+  }
+
+  static boolean isWindows(String os) {
+    return os != null && os.toLowerCase(Locale.ROOT).indexOf("win") > -1
+  }
+
+  static boolean isWindows() {
+    return isWindows(System.getProperty("os.name"))
+  }
+
+  static boolean createNewScriptFile(File outputFile) throws IOException {
```

Comment:
If this only throws, what is the point of this return code and the `if` in `createJarTrampolineScript()`?

Reply:
good point, will fix. at one point script writing was conditional only if the file was newly created ... but now unconditional to avoid stale scripts so return value isn't necessary.